File: DesignPatternsExercises/java/AdapterDesignPattern/Exercise/GUIAdapter.java
public class GUIAdapter implements __________(3)__________ {
	private WindowsGUI winGUI;
	private LinuxGUI linuxGUI;
	private MacGUI macGUI;

	public GUIAdapter(String operatingSystem) {
		if(operatingSystem.contains("Windows")) {
			winGUI = new WindowsGUI();
		} else if(operatingSystem.contains("Linux")) {
			linuxGUI = new LinuxGUI();
		} else if(operatingSystem.contains("Mac")) {
			macGUI = new MacGUI();
		} else {
			System.out.println("Unable to determine OS");
		}
	}

	@Override
	public void display() {
		if(winGUI != null) {
			__________(4)__________.display();
		} else if(linuxGUI != null) {
			__________(5)__________.display();
		} else if(macGUI != null) {
			__________(6)__________.display();
		} else {
			System.out.println("Unable to display GUI for unknown OS");
		}
	}
}

File: DesignPatternsExercises/ruby/adapter_design_pattern/solution/gui_adapter.rb
require_relative 'windows_gui'
require_relative 'mac_gui'
require_relative 'linux_gui'

class GUIAdapter
  # def __________(1)__________(operating_system)
  def initialize(operating_system)
    if operating_system.include? 'linux'
      # @os_gui = __________(2)__________
      @os_gui = LinuxGUI.new
      @display = 'display_linux'
    elsif operating_system.include? 'mac'
      @os_gui = MacGUI.new
      @display = 'display_mac'
    elsif operating_system.include? 'windows'
      @os_gui = WindowsGUI.new
      @display = 'display_windows'
    end
  end

  def display
    # @os_gui.send(__________(3)__________)
    @os_gui.send(@display)
  end
end
